ignite/contrib/handlers/tqdm_logger.py

```diff
@@ -10,47 +10,84 @@ class ProgressBar:
     """
     TQDM progress bar handler to log training progress and computed metrics.
 
+    Args:
+        persist (bool): (Optional) set to ``True`` to persist the progress bar after completion (default = ``False``)
+
     Examples:
 
-        Create a progress bar that shows you some metrics as they are computed,
-        by simply attaching the progress bar object to your engine.
+        Simple progress bar
+
+        .. code-block:: python
+
+            trainer = create_supervised_trainer(model, optimizer, loss)
+
+            pbar = ProgressBar()
+            pbar.attach(trainer)
+
+        Attach metrics that already have been computed at `ITERATION_COMPLETED` (such as `RunningAverage`)
 
         .. code-block:: python
 
+            trainer = create_supervised_trainer(model, optimizer, loss)
+
+            RunningAverage(output_transform=lambda x: x).attach(trainer, 'loss')
+
             pbar = ProgressBar()
             pbar.attach(trainer, ['loss'])
 
+        Directly attach the engine's output
+
+        .. code-block:: python
+
+            trainer = create_supervised_trainer(model, optimizer, loss)
+
+            pbar = ProgressBar()
+            pbar.attach(trainer, output_transform=lambda x: {'loss': x})
+
     Note:
         When adding attaching the progress bar to an engine, it is recommend that you replace
         every print operation in the engine's handlers triggered every iteration with
         ``pbar.log_message`` to guarantee the correct format of the stdout.
     """
 
-    def __init__(self):
+    def __init__(self, persist=False):
         self.pbar = None
+        self.persist = persist
 
     def _reset(self, engine):
         self.pbar = tqdm(
             total=len(engine.state.dataloader),
-            leave=False,
+            leave=self.persist,
             bar_format='{desc}[{n_fmt}/{total_fmt}] {percentage:3.0f}%|{bar}{postfix} [{elapsed}<{remaining}]')
 
     def _close(self, engine):
         self.pbar.close()
         self.pbar = None
 
-    def _update(self, engine, metric_names=None):
+    def _update(self, engine, metric_names=None, output_transform=None):
         if self.pbar is None:
             self._reset(engine)
 
         self.pbar.set_description('Epoch {}'.format(engine.state.epoch))
 
+        metrics = {}
         if metric_names is not None:
             if not all(metric in engine.state.metrics for metric in metric_names):
                 self._close(engine)
                 raise KeyError("metrics not found in engine.state.metrics")
 
-            metrics = {name: '{:.2e}'.format(engine.state.metrics[name]) for name in metric_names}
+            metrics.update({name: '{:.2e}'.format(engine.state.metrics[name]) for name in metric_names})
+
+        if output_transform is not None:
+            output_dict = output_transform(engine.state.output)
```

Comment:
IMO, to force the output as dict is a bit too strict and too verbose if user would like to display a single scalar, e.g. `loss.item()`. However this can be more interesting to display multiple losses without smoothing. Maybe we can let two possibilities for the output: dict and "not dict" which is then wrapped into a dict as : 
```python
output_dict = {"output": output_dict}
```

Reply:
it is harder to check for the proper output type, however. My bet would be to accept either a scalar or a `dict`

Reply:
Actually, you are not checking if the output is printable, I think this is OK. So you can just additionally compare if the output is not dict and wrap it to the dict.

Reply:
> > what do you mean by that?
> 
> @miguelvr take a look [here](https://github.com/pytorch/ignite/issues/296#issuecomment-431697231) at display custom variables.
> 
> Actually, I think we will need to do a similar stuff for RunningAverage to support custom variables too...

I did look at that, but I don't understand the usage of it. Can you give me an example of it being necessary?